registry/storage/walk.go

```diff
@@ -38,9 +43,16 @@ func Walk(ctx context.Context, driver storageDriver.StorageDriver, from string,
 			return err
 		}
 		err = f(fileInfo)
-		skipDir := (err == ErrSkipDir)
-		if err != nil && !skipDir {
-			return err
+
+		skipDir := false
+		if err != nil {
+			if err == ErrFinishedWalk {
+				break
+			} else if err == ErrSkipDir {
+				skipDir = true
+			} else {
+				return err
```

Comment:
The nil return should pass back a nil and terminate the Walk(), which isn't quite the same as skipDir here, but your point is valid;  I can just modify the called walk function and have it pass back any old error to get the same result.